lib/beaker/testmode_switcher/beaker_runners.rb

```diff
@@ -15,6 +15,34 @@ def initialize(hosts, logger)
         self.logger = logger
       end
 
+      # This method will return an array of acceptable puppet run (agent or apply)
+      # exit codes based on the expectations laid out by given options. you may
+      # specify only one of the below options in the opts param
+      # Options:
+      # :catch_changes
+      # :catch_failures
+      # :expect_changes
+      # :expect_failures
+      def get_acceptable_puppet_run_exit_codes(opts = {})
+        # Ensure only one option given
+        if [opts[:catch_changes], opts[:catch_failures], opts[:expect_failures], opts[:expect_changes]].compact.length > 1
+          raise(ArgumentError,
+                'Cannot specify more than one of `catch_failures`, \
+                `catch_changes`, `expect_failures`, or `expect_changes` \
+                 for a single manifest')
+        end
+
+        # Return appropriate exit code
+        return [0] if opts[:catch_changes]
+        return [0, 2] if opts[:catch_failures]
+        return [1, 4, 6] if opts[:expect_failures]
+        return [2] if opts[:expect_changes]
+
+        # If no option supplied, return all exit codes, as an array,
+        # as acceptable so beaker returns a detailed output
+        (0...256).to_a
```

Comment:
no `.to_a` needed here. Ranges and Arrays are interchangeable in most situations in ruby. In this specific context it will only waste memory and cpu.